src/core/include/openvino/runtime/tensor_util.hpp

```diff
@@ -0,0 +1,89 @@
+#endif
+OPENVINO_API
+void save_tensor_data(const Tensor& tensor, const std::filesystem::path& file_name);
+
+/// \brief Read a tensor content from a file. Only raw data is loaded.
+/// \param file_name Path to file to read.
+/// \param element_type Element type, when not specified the it is assumed as element::u8.
+/// \param shape Shape for resulting tensor. If provided shape is static, specified number of elements is read only.
+///              File should contain enough bytes, an exception is raised otherwise.
+///              One of the dimensions can be dynamic. In this case it will be determined automatically based on the
+///              length of the file content and `offset`. Default value is [?].
+/// \param offset Read file starting from specified offset. Default is 0. The remining size of the file should be
+/// compatible with shape.
+/// \param mmap Use mmap that postpones real read from file until data is accessed. If mmap is used, the file
+///             should not be modified until returned tensor is destroyed.
+OPENVINO_API
+Tensor read_tensor_data(const std::string& file_name,
+                        const element::Type& element_type = element::u8,
+                        const PartialShape& shape = PartialShape{Dimension::dynamic()},
+                        std::size_t offset = 0,
+                        bool mmap = true);
+
+#if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT)
+OPENVINO_API
+Tensor read_tensor_data(const std::wstring& file_name,
+                        const element::Type& element_type = element::u8,
+                        const PartialShape& shape = PartialShape{Dimension::dynamic()},
+                        std::size_t offset = 0,
+                        bool mmap = true);
+#endif
+OPENVINO_API
+Tensor read_tensor_data(const std::filesystem::path& file_name,
+                        const element::Type& element_type = element::u8,
+                        const PartialShape& shape = PartialShape{Dimension::dynamic()},
+                        std::size_t offset = 0,
+                        bool mmap = true);
+
+/// \brief Read raw data from a file into pre-allocated tensor.
+/// \param file_name Path to the input file with raw tensor data.
+/// \param tensor Tensor to read data to. Tensor should have correct element_type and shape set that is used to
+/// determine how many bytes will be read from the file.
+/// \param offset Read file starting from specified offset. Default
+/// is 0. The remining part of the file should contain enough bytes to satisfy tensor size.
+OPENVINO_API
+void read_tensor_data(const std::string& file_name, Tensor& tensor, std::size_t offset = 0);
+#if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT)
+OPENVINO_API
+void read_tensor_data(const std::wstring& file_name, Tensor& tensor, std::size_t offset = 0);
+#endif
+OPENVINO_API
+void read_tensor_data(const std::filesystem::path& file_name, Tensor& tensor, std::size_t offset = 0);
```

Comment:
Keep only this version to use path type only. 
In future the it should be only way to pass paths instead string/wstring

Mayeb these functions can be static members of Tensor class? and use it like
```auto tensor = ov::Tensor::read(file_path, offset = 0);```

Reply:
only std::filesystem::path kept

answered to anther comment